home-service/src/main/java/com/softserveinc/ita/homeproject/homeservice/service/impl/ContactServiceImpl.java

```diff
@@ -0,0 +1,104 @@
+package com.softserveinc.ita.homeproject.homeservice.service.impl;
+
+import java.util.Optional;
+
+import com.softserveinc.ita.homeproject.homedata.entity.Contact;
+import com.softserveinc.ita.homeproject.homedata.entity.ContactType;
+import com.softserveinc.ita.homeproject.homedata.entity.Email;
+import com.softserveinc.ita.homeproject.homedata.entity.Phone;
+import com.softserveinc.ita.homeproject.homedata.entity.User;
+import com.softserveinc.ita.homeproject.homedata.repository.ContactRepository;
+import com.softserveinc.ita.homeproject.homeservice.dto.ContactDto;
+import com.softserveinc.ita.homeproject.homeservice.dto.EmailContactDto;
+import com.softserveinc.ita.homeproject.homeservice.dto.PhoneContactDto;
+import com.softserveinc.ita.homeproject.homeservice.exception.NotFoundHomeException;
+import com.softserveinc.ita.homeproject.homeservice.mapper.ServiceMapper;
+import com.softserveinc.ita.homeproject.homeservice.service.ContactService;
+import com.softserveinc.ita.homeproject.homeservice.service.UserService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class ContactServiceImpl implements ContactService {
+
+    private final ContactRepository contactRepository;
+
+    private final ServiceMapper mapper;
+
+    private final UserService userService;
+
+    @Override
+    public ContactDto createContact(Long userId, ContactDto createContactDto) {
+        Contact contact = mapper.convert(createContactDto, Contact.class);
+        User user = mapper.convert(userService.getUserById(userId), User.class);
+        contact.setUser(user);
+        contactRepository.save(contact);
+        return mapper.convert(contact, ContactDto.class);
+    }
+
+    @Transactional
```

Comment:
Do we really need a transaction here?

Reply:
We needed it before because it didn't work without it but now it works fine without this annotation so I deleted it. Thanks for paying attendtion on it.